src/RedirectMiddleware.php

```diff
@@ -214,9 +214,20 @@ private function redirectUri(
         ResponseInterface $response,
         array $protocols
     ) {
+        $locationHeaderLine = $response->getHeaderLine('Location');
+
+        // Avoid TooManyRedirectsException when following empty location headers
+        if (empty($locationHeaderLine)) {
+            throw new BadResponseException(
+                'The server responded with a 302 status code, but supplied an empty Location header',
```

Comment:
This seems broken:
1. it does not need to be a 302, it could also be any other 3xx status code
2. An empty location header would be valid for a PUT request that returns a 303 with an empty location header, It means the resource was created and you can GET it from the same URI.
3. A 3xx response usually can also contain a body which a client might want to consume. Throwing an exception seems very unfriendly when the empty Location is just a minor error that could be ignored. If we throw an exception it needs to take the https://github.com/guzzle/guzzle/blob/master/src/RequestOptions.php#L133 option into account.

I think it makes sense to handle this case but needs some more work. Maybe just excluding 303 to solve 2) is enough.

Reply:
Ok, thanks! Let me have a look tot fix this.

Reply:
@Tobion I changed the exception message, but the exception should actually be removed and solved in another way.

The way you describe PUT requests with empty locations, means that Guzzle should auto fill the response Location header with the request URI.

This could still end up with a `TooManyRedirectsException` if the same uri keeps returning the same response.